cdap-api/src/main/java/co/cask/cdap/internal/schedule/trigger/ScheduleTrigger.java

```diff
@@ -0,0 +1,609 @@
+      super(Type.AND, triggers);
+    }
+  }
+
+  /**
+   * A Trigger builder that builds a {@link AndTrigger}.
+   */
+  public static class OrTriggerBuilder extends AbstractCompositeTrigger implements TriggerBuilder {
+
+    public OrTriggerBuilder(Trigger... triggers) {
+      super(Type.OR, triggers);
+    }
+
+    @Override
+    public Trigger build(String namespace, String applicationName, String applicationVersion) {
+      int numTriggers = getTriggers().length;
+      Trigger[] builtTriggers = new Trigger[numTriggers];
+      for (int i = 0; i < numTriggers; i++) {
+        Trigger trigger = getTriggers()[i];
+        builtTriggers[i] = trigger instanceof TriggerBuilder ?
+          (ScheduleTrigger) ((TriggerBuilder) trigger).build(namespace, applicationName, applicationVersion) : trigger;
+      }
+      return new OrTrigger(builtTriggers);
+    }
+  }
+
+  /**
+   * Represents an OR trigger in an app.
+   */
+  public static class OrTrigger extends AbstractCompositeTrigger {
+    public OrTrigger(Trigger... triggers) {
+      super(Type.OR, triggers);
+    }
+  }
+
+  /**
+   * Represents a stream size trigger in an app.
+   */
+  public static class StreamSizeTrigger extends ScheduleTrigger {
+
+    private final String streamNamespace;
+    private final String streamName;
+    private final int triggerMB;
+
+    public StreamSizeTrigger(String streamNamespace, String streamName, int triggerMB) {
+      super(Type.STREAM_SIZE);
+      this.streamNamespace = streamNamespace;
+      this.streamName = streamName;
+      this.triggerMB = triggerMB;
+      validate();
+    }
+
+    public String getStreamNamespace() {
+      return streamNamespace;
+    }
+
+    public String getStreamName() {
+      return streamName;
+    }
+    
+    public int getTriggerMB() {
+      return triggerMB;
+    }
+
+    @Override
+    public void validate() {
+      validateNotNull(getStreamNamespace(), "stream namespace");
+      validateNotNull(getStreamName(), "stream name");
+      validateInRange(getTriggerMB(), "trigger in MB", 1, null);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      return this == o ||
+        o != null &&
+          getClass().equals(o.getClass()) &&
+          Objects.equals(getStreamNamespace(), ((StreamSizeTrigger) o).getStreamNamespace()) &&
+          Objects.equals(getStreamName(), ((StreamSizeTrigger) o).getStreamName()) &&
+          Objects.equals(getTriggerMB(), ((StreamSizeTrigger) o).getTriggerMB());
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(getStreamNamespace(), getStreamName(), getTriggerMB());
+    }
+
+    @Override
+    public String toString() {
+      return String.format("StreamSizeTrigger(%s.%s, %d MB)", getStreamNamespace(), getStreamName(), getTriggerMB());
+    }
+  }
+
+  /**
+   * A Trigger builder that builds a {@link StreamSizeTrigger}.
+   */
+  public static class StreamSizeTriggerBuilder extends ScheduleTrigger implements TriggerBuilder {
```

Comment:
Why the build is needed in `cdap-api`? I thought it is only used by the configurer?